### ISSUE TITLE
Error handing is a little broken

### ISSUE THREAD

**Issue by SuperDisk** [Rob-bie/ElixirPlusReddit#2]
Whenever an error occurs (for instance posting comments more often than allowed), ElixirPlusReddit crashes. I'll post a stack trace tomorrow.

**Comment by Rob-bie** [comment]
Yeah, EPR is incomplete as a whole. I never implemented a strategy for handling this sort of rate limiting. This sort of rate limiting is very variable and is based on things like karma, what subreddit you're posting to and so forth.

**Comment by SuperDisk** [comment]
Well, regardless of whether it's finished or not, it was pretty invaluable in my Reddit bot project. Seriously thanks for making it!

**Comment by Rob-bie** [comment]
And thank you for using it. I was planning on starting an 'EPR2' with improved design, better OAuth2 support, hitting more endpoints, etc, but have yet to get around to it. I was actually looking for people to start over the project with.

**Comment by SuperDisk** [comment]
Maybe we could team up. However I will say I pretty much just started with Elixir and don't know the best practices, so if you're shooting for better design then maybe not.

**Comment by egze** [comment]
@Rob-bie What would you change in the design? Are you still thinking about doing EPR2? I am interested.
